_posts/2022-06-27-changelog-173.markdown

```diff
@@ -0,0 +1,49 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #173"
+date: 2022-06-27 13:00:00 +0300
+---
+
+
+## New Features
+
+* [#8927] Take into account [`#![recursion_limit]`](https://doc.rust-lang.org/reference/attributes/limits.html#the-recursion_limit-attribute)  which controls the maximum depth of macro expansions. This fixes some unresolved references when using such crates as `async-std` and `tokio`. If you don't want/need to expand macros to full depth, you can adjust "Maximum recursion limit for macro expansion" in [Advanced Settings](https://jetbrains.com/help/idea/advanced-settings.html).
+
+## Performance Improvements
+
+* [#8955] Reduce memory usage by `TraitImplSource` instances since they're extensively cached and thus bloat old generation heap space a lot
+
+## Fixes
+
+* [#8957] Take into account impls for macro types. For example, `impl foo!() { fn bar(&self) {} }`
+
+* [#8952] Complete more methods when autoderef occurs
+
+* [#8950] Consider methods in impls for a reference (`&Foo`) when the receiver is not a reference type
+
+* [#8943] Consider [negative impls](https://github.com/rust-lang/rust/issues/68318) (e.g. `impl !Sync for T {}`) in type inference
+
+* [#8934] The  "virtual" `clippy::all` lint group will now be completed (by [@Kobzol])
+
+## Internal Improvements
+
+* [#8919] Implement a new approach to better detect changes that possibly affect the project model. It's disabled by default for now.
+
+* [#8900] Remove [Old Macro Expansion Engine](https://blog.jetbrains.com/clion/2020/07/intellij-rust-0-3-new-macro-expansion-engine) option
```

Comment:
It's visible for users, so it's not an internal change. Let's move to `feature` section